Yesod/Paginator/Widget.hs

```diff
@@ -115,12 +117,20 @@ paginationWidget (PageWidgetConfig {..}) page per tot = do
 
                 -- the middle lists, strip the first/last pages and
                 -- correctly take up to limit away from current
-                prevLinks = reverse . take pageCount . reverse . drop 1 $ map (\p -> Enabled p (showT p) "prev") prev
-                nextLinks = take pageCount . reverse . drop 1 . reverse $ map (\p -> Enabled p (showT p) "next") next
+                prevLinks = reverse . take pageCount . reverse . drop 1 $ map (display "prev") prev
+                nextLinks = take pageCount . reverse . drop 1 . reverse $ map (display "next") next
 
                 -- finally, this page itself
                 curLink = [Disabled (showT pg) "active"]
 
+                pageWithinBounds p = case maxPageLinks of
+                    Nothing -> True
+                    Just m  -> if p > m then False
+                                       else True
+                display cls p = if (pageWithinBounds p)
```

Comment:
Is there a better name for this? I think it's more than just "display"; it's actually "display if within bounds".